src/Expander.hs

```diff
@@ -1153,3 +1373,215 @@ interpretMacroAction (MacroActionIdentEq how v1 v2) = do
 interpretMacroAction (MacroActionLog stx) = do
   liftIO $ prettyPrint stx >> putStrLn ""
   pure $ Right (ValueBool False) -- TODO unit type
+
+-- | Invariant: the SplitCorePtr points at the CoreF in question
+typeCheckLayer :: SplitCorePtr -> CoreF SplitCorePtr -> Ty -> Expand ()
+typeCheckLayer dest (CoreVar _) t =
+  forkCheckVar dest t
+typeCheckLayer dest (CoreLet x ident def body) t = do
+  xTy <- inTypeBinder do
```

Comment:
```suggestion
  xTy <- inTypeBinder $ do
```

Reply:
Luckily it's 2020 and we have `-XBlockArguments`, so there's no need for an infix application operator to say "here comes a `do`!".